scene/main/window.cpp

```diff
@@ -665,10 +665,15 @@ void Window::_make_window() {
 		window_rect = Rect2i(DisplayServer::get_singleton()->screen_get_position(DisplayServer::SCREEN_WITH_KEYBOARD_FOCUS) + (DisplayServer::get_singleton()->screen_get_size(DisplayServer::SCREEN_WITH_KEYBOARD_FOCUS) - size) / 2, size);
 	}
 
-	window_id = DisplayServer::get_singleton()->create_sub_window(DisplayServer::WindowMode(mode), vsync_mode, f, window_rect, is_in_edited_scene_root() ? false : exclusive, transient_parent ? transient_parent->window_id : DisplayServer::INVALID_WINDOW_ID);
+	// Add parent window if --swid told us to embed a window with this title.
+	int64_t parent_handle = Engine::get_singleton()->get_embedded_subwindow(get_title());
```

Comment:
I would use something more unique, like node path (`get_path()`) to identify windows (title still can be shown in the user facing UI).